phplib/admin-reps.php

```diff
@@ -239,442 +259,513 @@ function display()
-                if ($repinfo['parlparse_person_id']) {
-                    $sameperson = dadem_get_same_person($repinfo['parlparse_person_id']);
-                    dadem_check_error($sameperson);
-                }
-            }
-            $va_id = $rep_id ? $repinfo['voting_area'] : $new_in_va_id;
-            $vainfo = mapit_call('area', $va_id);
-            mapit_check_error($vainfo);
-            if ($vainfo['parent_area']) {
-                $parentinfo = mapit_call('area', $vainfo['parent_area']);
-                mapit_check_error($parentinfo);
-            } else {
-                $parentinfo = null;
-            }
-            $rephistory = $rep_id ? dadem_get_representative_history($rep_id) : array();
-            dadem_check_error($rephistory);
-            // Reverse postcode lookup
-            if (!$pc) {
-                $pc = mapit_call('area/example_postcode', $va_id);
-                if (!mapit_get_error($pc)) {
-                    $form->addElement('static', 'note1', null, "Example postcode for testing: " .
-                        "<a href='" . OPTION_BASE_URL . '/who?pc=' . urlencode($pc) . "'>"
-                        . htmlentities($pc) ."</a> (<a href='?search=" . urlencode($pc)
-                        . "&amp;gos=postcode+or+query&amp;page=reps'>all reps here</a>)");
-                } else {
-                    $pc = '';
-                }
+        if ($this->params['rep_id'] or $this->params['new_in_va_id']) {
+            $this->displayRep();
+        } elseif ($this->params['va_id']) {
+            $this->displayVotingArea();
+        } elseif ($this->params['search']) {
+            $this->displaySearch();
+        } elseif ($this->params['pc']) {
+            $this->displayPostcode();
+        } elseif (get_http_var('bad_contacts')) {
+            $this->displayBadContacts();
+        } elseif (get_http_var('user_corrections')) {
+            $this->displayUserCorrections();
+        } else {
+            print '<p><a href="?page=reps&bad_contacts=1">Bad contacts</a> (please fix these!)';
+            // General info
+            if (OPTION_ADMIN_SERVICES_CGI) {
+                print '<br><a href="?page=reps&user_corrections=1">User corrections</a>
+                    (just for your interest, as sent automatically to GovEval)';
             }
+        }
+    }
 
-            if ($rep_id) {
-                $form->setDefaults(
-                    array('name' => $repinfo['name'],
-                    'party' => $repinfo['party'],
-                    'method' => $repinfo['method'],
-                    'email' => $repinfo['email'],
-                    'fax' => $repinfo['fax'])
-                );
-            }
+    private function getReps($va_id)
+    {
+        $reps = dadem_get_representatives($va_id);
+        dadem_check_error($reps);
+        $reps = array_values($reps);
+        return $reps;
+    }
+
+    private function displayRepAreaInfo($va_id)
```

Comment:
Yeah, all the displayRep related functions start displayRep, so I guess it should be displayRepGetAreaInfo?